misc/buildkite/git.bash

```diff
@@ -13,8 +13,13 @@ set -euo pipefail
 
 . misc/shlib/shlib.bash
 
-export BUILDKITE_REPO_REF="${BUILDKITE_REPO_REF:-origin}"
-export BUILDKITE_PULL_REQUEST_BASE_BRANCH="${BUILDKITE_PULL_REQUEST_BASE_BRANCH:-main}"
+MZ_REPO_REF="${BUILDKITE_REPO_REF:-origin}"
+MZ_REPO_PULL_REQUEST_BASE_BRANCH="${BUILDKITE_PULL_REQUEST_BASE_BRANCH:-main}"
+
+if [[ "${BUILDKITE:-}" != "true" ]]; then
+  # when running locally, we origin may point to a fork of the repo but we want the mz repo
+  MZ_REPO_REF=$(git remote -v | grep "MaterializeInc/materialize" | grep "fetch" | cut -f1)
```

Comment:
Someone could have the repo twice:
```
[remote "origin"]
	url = git@github.com:def-/materialize
	fetch = +refs/heads/*:refs/remotes/origin/*
[remote "upstream2"]
	url = git@github.com:MaterializeInc/materialize
	fetch = +refs/heads/*:refs/remotes/upstream2/*
```
Then there are two results here. I'd suggest to not try to parse the remote at all.
```suggestion
  MZ_REPO_REF="git@github.com:MaterializeInc/materialize"
```

Reply:
`git merge-base HEAD "$MZ_REPO_REF"/main` does not work:
```
fatal: Not a valid object name git@github.com:MaterializeInc/materialize/main
```

What I could do is to pick only the first line. This will still fail if the mz repo has not been added as a remote but not sure how relevant that is.

Reply:
Amended the command with `| head -n1`.